### PR TITLE
More cleanups

### PR DESCRIPTION
**Pull request by zx2c4**
Automatically getting the list of all repos is particularly nice in this series.

### CODE REVIEW

File: etc/portage/postsync.d/sync_cache [hunk 1]
```diff
@@ -4,20 +4,12 @@ source /etc/portage/check-portdir.sh
...
+rsync -aq rsync://rsync.gentoo.org/gentoo-portage/metadata/md5-cache/ "${PORTDIR}"/metadata/md5-cache/
 eend $?
 
-if [[ ${#overlay_list[@]} -gt 0 ]] ; then
-	ebegin "Syncing metadata cache for overlays"
-	for _overlay in ${overlay_list[@]} ; do
-		einfo "  ${_overlay}"
-		egencache --jobs=$(nproc) --repo=${_overlay} --update --update-use-local-desc
-	done
+for repo in $(sed -n '/^\[DEFAULT\]$/d;s/^\[\(.*\)\]$/\1/p' /etc/portage/repos.conf/*); do
```
Comment (hasufell) [inline etc/portage/postsync.d/sync_cache]: why is `/etc/portage` hardcoded here?

Comment (hasufell) [inline etc/portage/postsync.d/sync_cache]: what is that sed line even for?

You can use `portageq get_repos /`

**Comment by hasufell** [comment]
I personally am not sure this is a good idea. I have done something similar for myself in paludis, but I'm not sure if that is a good default. I'd rather add a comment like:

``` sh
# add your overlays here (name of the overlay!)
# overlay_list=( )

# if you want to update cache for all overlays, uncomment the following line
# overlay_list=( $(portageq get_repos / | sed 's#\(^\| \)gentoo\( \|$\)##') )
```

**Comment by zx2c4** [comment]
I disagree. I think doing it for all repos is indeed a sensible default. There is never any harm in generating the cache, and it only helps speed things up, since all emerge operations wind up using it. If users have very odd particular desires to disable generation of the cache, then they can make some modifications. Requiring users to edit the script for ordinary functionality seems a bit baroque.

I made your suggested change of portageq; thanks for that.

**Comment by hasufell** [comment]
> Requiring users to edit the script for ordinary functionality seems a bit baroque.

That's called configuration.

Anyway, I'd be fine to make it

``` sh
# If you don't want to update cache for all overlays, comment the following line or
# adjust it to your needs.
overlay_list=( $(portageq get_repos / | sed 's#\(^\| \)gentoo\( \|$\)##') )
```

but there is no good reason to remove that variable altogether.

**Comment by hasufell** [comment]
Or probably better:

``` sh
# by default syncs all available repositories
repository_list= ( $(portageq get_repos /) )
```

**Comment by hasufell** [comment]
only added the first commit

**Pull request closed without merging**